src/extensions/cookie/bootstrap-table-cookie.js

```diff
@@ -479,10 +498,32 @@ $.BootstrapTable = class extends $.BootstrapTable {
       throw new Error('Could not parse the json of the columns cookie!', columnsCookieValue)
     }
 
-    // sortOrder
-    this.options.sortOrder = sortOrderCookie ? sortOrderCookie : this.options.sortOrder
-    // sortName
-    this.options.sortName = sortOrderNameCookie ? sortOrderNameCookie : this.options.sortName
+    try {
+      sortPriorityCookie = JSON.parse(sortPriorityCookie)
+    } catch (e) {
+      throw new Error('Could not parse the json of the sortPriority cookie!', sortPriorityCookie)
+    }
+
+    if (!sortPriorityCookie) {
+      // sortOrder
+      this.options.sortOrder = sortOrderCookie ? sortOrderCookie : this.options.sortOrder
+      // sortName
+      this.options.sortName = sortOrderNameCookie ? sortOrderNameCookie : this.options.sortName
+    } else {
+      // sortOrder
+      this.options.sortOrder = undefined
+      // sortName
+      this.options.sortName = undefined
+    }
+
+    if (this.options.sortOrder || this.options.sortName) {
+      // sortPriority
+      this.options.sortPriority = null
+    } else {
+      // sortPriority
+      this.options.sortPriority = sortPriorityCookie ? sortPriorityCookie : this.options.sortPriority
```

Comment:
Ok for the "if"
But the check is important, otherwise, like the example, there can be a default sort in the table settings that will override the cookie